lib/AST/DiagnosticEngine.cpp

```diff
@@ -1011,9 +1084,15 @@ void DiagnosticEngine::emitDiagnostic(const Diagnostic &diagnostic) {
 
 const char *DiagnosticEngine::diagnosticStringFor(const DiagID id,
                                                   bool printDiagnosticName) {
+  // TODO: Print diagnostic names from `localization`.
   if (printDiagnosticName) {
     return debugDiagnosticStrings[(unsigned)id];
   }
+  if (localization.get()) {
```

Comment:
I think `if(localization)` should be fine, `std::unique_ptr` implements `bool()`